<a href="https://colab.research.google.com/github/RahulVerma2444/AI-ML_GroupProject/blob/main/Airline_Sentiment_Analysis_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import nltk
from nltk.tokenize import word_tokenize
from nltk.stem import WordNetLemmatizer
from nltk.stem import PorterStemmer
nltk.download('averaged_perceptron_tagger_eng')
nltk.download('wordnet')
nltk.download('omw-1.4')
nltk.download('punkt_tab')
nltk.download('stopwords')

[nltk_data] Downloading package averaged_perceptron_tagger_eng to
[nltk_data]     /root/nltk_data...
[nltk_data]   Unzipping taggers/averaged_perceptron_tagger_eng.zip.
[nltk_data] Downloading package wordnet to /root/nltk_data...
[nltk_data] Downloading package omw-1.4 to /root/nltk_data...
[nltk_data] Downloading package punkt_tab to /root/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt_tab.zip.
[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Unzipping corpora/stopwords.zip.


True

In [ ]:
import re

In [ ]:
!pip install emoji
import emoji

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 590.6/590.6 kB 6.0 MB/s eta 0:00:00


In [ ]:
import pandas as pd
from sklearn.model_selection import train_test_split

df = pd.read_csv("Tweets.csv")
df.head()

tweet_id airline_sentiment  airline_sentiment_confidence  \
0  570306133677760513           neutral                        1.0000   
1  570301130888122368          positive                        0.3486   
2  570301083672813571           neutral                        0.6837   
3  570301031407624196          negative                        1.0000   
4  570300817074462722          negative                        1.0000   

  negativereason  negativereason_confidence         airline  \
0            NaN                        NaN  Virgin America   
1            NaN                     0.0000  Virgin America   
2            NaN                        NaN  Virgin America   
3     Bad Flight                     0.7033  Virgin America   
4     Can't Tell                     1.0000  Virgin America   

  airline_sentiment_gold        name negativereason_gold  retweet_count  \
0                    NaN     cairdin                 NaN              0   
1                    NaN    jnardino                 NaN              0   
2                    NaN  yvonnalynn                 NaN              0   
3                    NaN    jnardino                 NaN              0   
4                    NaN    jnardino                 NaN              0   

                                                text tweet_coord  \
0                @VirginAmerica What @dhepburn said.         NaN   
1  @VirginAmerica plus you've added commercials t...         NaN   
2  @VirginAmerica I didn't today... Must mean I n...         NaN   
3  @VirginAmerica it's really aggressive to blast...         NaN   
4  @VirginAmerica and it's a really big bad thing...         NaN   

               tweet_created tweet_location               user_timezone  
0  2015-02-24 11:35:52 -0800            NaN  Eastern Time (US & Canada)  
1  2015-02-24 11:15:59 -0800            NaN  Pacific Time (US & Canada)  
2  2015-02-24 11:15:48 -0800      Lets Play  Central Time (US & Canada)  
3  2015-02-24 11:15:36 -0800            NaN  Pacific Time (US & Canada)  
4  2015-02-24 11:14:45 -0800            NaN  Pacific Time (US & Canada)

In [ ]:
y = df['airline_sentiment']
X = df['text']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


In [ ]:
print(y_train)

750      negative
6875     negative
7598     positive
14124    negative
6187      neutral
           ...   
5191     positive
13418    negative
5390     negative
860      negative
7270      neutral
Name: airline_sentiment, Length: 11712, dtype: object


In [ ]:
punctuation = [',', '.', ':', ';', '!', '\"', '?']

In [ ]:
def has_emoji(text):
  for char in text:
    if emoji.is_emoji(char):
      return True
  return False

In [ ]:
count1 = 0
count2 = 0
count3 = 0
count4 = 0
i = 0

print('Negative')
for data in X_train:
  if '@' in data and count1 < 10 and y_train.iloc[i] == 'negative':
    print("\nCONTAINS @")
    print(data)
    #print(y_train[count1])
    count1 += 1
  if ('http' in data or 'https' in data) and count2 < 10 and y_train.iloc[i] == 'negative':
    print("\nhttp or https")
    print(data)
    #print(y_train[count1])
    count2 += 1
  if any(p in data for p in punctuation) and count3 < 10 and y_train.iloc[i] == 'negative':
    print("\nPUNCTUATION")
    print(data)
    #print(y_train[count1])
    count3 += 1
  if has_emoji(data) and count4 < 10 and y_train.iloc[i] == 'negative':
    print("\nEMOJI")
    print(data)
    count4 += 1
  i += 1

Negative

CONTAINS @
@united you are offering us 8 rooms for 32 people #FAIL

CONTAINS @
@JetBlue JFK NYC staff is amazing. The #lax JetBlue... Sending an email with details but it was a disappointing experience @JetBlueCheeps

PUNCTUATION
@JetBlue JFK NYC staff is amazing. The #lax JetBlue... Sending an email with details but it was a disappointing experience @JetBlueCheeps

CONTAINS @
@AmericanAir flight 3056 still sitting at DFW waiting for baggage to be loaded

CONTAINS @
@united   Pushing 2 hours on hold. Priceless. http://t.co/thS10LDY2a

http or https
@united   Pushing 2 hours on hold. Priceless. http://t.co/thS10LDY2a

PUNCTUATION
@united   Pushing 2 hours on hold. Priceless. http://t.co/thS10LDY2a

CONTAINS @
@AmericanAir I've been trying to reach your customer service for TWO days. I have received zero response. Never traveling AA again.

PUNCTUATION
@AmericanAir I've been trying to reach your customer service for TWO days. I have received zero response. Never traveling AA ag

In [ ]:
count1 = 0
count2 = 0
count3 = 0
count4 = 0
i = 0

print('Neutral')
for data in X_train:
  if '@' in data and count1 < 10 and y_train.iloc[i] == 'neutral':
    print("\nCONTAINS @")
    print(data)
    #print(y_train[count1])
    count1 += 1
  if ('http' in data or 'https' in data) and count2 < 10 and y_train.iloc[i] == 'neutral':
    print("\nhttp or https")
    print(data)
    #print(y_train[count1])
    count2 += 1
  if any(p in data for p in punctuation) and count3 < 10 and y_train.iloc[i] == 'neutral':
    print("\nPUNCTUATION")
    print(data)
    #print(y_train[count1])
    count3 += 1
  if has_emoji(data) and count4 < 10 and y_train.iloc[i] == 'neutral':
    print("\nEMOJI")
    print(data)
    count4 += 1
  i += 1

Neutral

CONTAINS @
@southwestair is your companion pass broken today? 
purchase.error.INVALID_LOYALTY_MEMBER_ACCOUNT_STATUS (SW900001-vyiL1XKlROG24fS-918j_A)

PUNCTUATION
@southwestair is your companion pass broken today? 
purchase.error.INVALID_LOYALTY_MEMBER_ACCOUNT_STATUS (SW900001-vyiL1XKlROG24fS-918j_A)

CONTAINS @
@USAirways Don't want to clog your already-crowded line with a special meal request. Can someone DM me to sort out? #vegetarianproblems :)

PUNCTUATION
@USAirways Don't want to clog your already-crowded line with a special meal request. Can someone DM me to sort out? #vegetarianproblems :)

CONTAINS @
@JetBlue what is the Mosiac 800 #?

PUNCTUATION
@JetBlue what is the Mosiac 800 #?

CONTAINS @
@united - sitting in seat 10D on a flight back from Vegas to Chicago.  Someone left a folder that looks important in the seat pocket. (1/2)

PUNCTUATION
@united - sitting in seat 10D on a flight back from Vegas to Chicago.  Someone left a folder that looks important in the seat 

In [ ]:
count1 = 0
count2 = 0
count3 = 0
count4 = 0
i = 0

print('Positive')
for data in X_train:
  if '@' in data and count1 < 10 and y_train.iloc[i] == 'positive':
    print("\nCONTAINS @")
    print(data)
    #print(y_train[count1])
    count1 += 1
  if ('http' in data or 'https' in data) and count2 < 10 and y_train.iloc[i] == 'positive':
    print("\nhttp or https")
    print(data)
    #print(y_train[count1])
    count2 += 1
  if any(p in data for p in punctuation) and count3 < 10 and y_train.iloc[i] == 'positive':
    print("\nPUNCTUATION")
    print(data)
    #print(y_train[count1])
    count3 += 1
  if has_emoji(data) and count4 < 10 and y_train.iloc[i] == 'positive':
    print("\nEMOJI")
    print(data)
    count4 += 1
  i += 1

Positive

CONTAINS @
@JetBlue well, the last update was in the right direction, at least! I'll keep my fingers crossed.

PUNCTUATION
@JetBlue well, the last update was in the right direction, at least! I'll keep my fingers crossed.

CONTAINS @
@united that's great! pls let me know when u start!

PUNCTUATION
@united that's great! pls let me know when u start!

CONTAINS @
@SouthwestAir please acknowledge Attendant Jeff Wernicke. Beat in class and service on SW

PUNCTUATION
@SouthwestAir please acknowledge Attendant Jeff Wernicke. Beat in class and service on SW

CONTAINS @
@SouthwestAir really appreciate the follow up, I always fly with y'all for a reason!

PUNCTUATION
@SouthwestAir really appreciate the follow up, I always fly with y'all for a reason!

CONTAINS @
@SouthwestAir yes I was everything worked out great

CONTAINS @
@AmericanAir thank you! My lost item was located at ORD and is being returned to me.

PUNCTUATION
@AmericanAir thank you! My lost item was located at ORD and is be

In [ ]:
tokens = ['@', 'http', 'https', ',', '.', ':', ';', '!', '\"', '?']
def RemoveTokens(text):
  text = word_tokenize(text)

  text = [word for word in text if word not in tokens]
  text = " ".join(text)
  text = emoji.replace_emoji(text, replace='')

  return text

cleaned_data = []
for data in X:
  data1 = RemoveTokens(data)
  cleaned_data.append(data1)

df['cleaned_data'] = cleaned_data
print(df)

                 tweet_id airline_sentiment  airline_sentiment_confidence  \
0      570306133677760513           neutral                        1.0000   
1      570301130888122368          positive                        0.3486   
2      570301083672813571           neutral                        0.6837   
3      570301031407624196          negative                        1.0000   
4      570300817074462722          negative                        1.0000   
...                   ...               ...                           ...   
14635  569587686496825344          positive                        0.3487   
14636  569587371693355008          negative                        1.0000   
14637  569587242672398336           neutral                        1.0000   
14638  569587188687634433          negative                        1.0000   
14639  569587140490866689           neutral                        0.6771   

               negativereason  negativereason_confidence         airline  \

In [ ]:
df.head()

tweet_id airline_sentiment  airline_sentiment_confidence  \
0  570306133677760513           neutral                        1.0000   
1  570301130888122368          positive                        0.3486   
2  570301083672813571           neutral                        0.6837   
3  570301031407624196          negative                        1.0000   
4  570300817074462722          negative                        1.0000   

  negativereason  negativereason_confidence         airline  \
0            NaN                        NaN  Virgin America   
1            NaN                     0.0000  Virgin America   
2            NaN                        NaN  Virgin America   
3     Bad Flight                     0.7033  Virgin America   
4     Can't Tell                     1.0000  Virgin America   

  airline_sentiment_gold        name negativereason_gold  retweet_count  \
0                    NaN     cairdin                 NaN              0   
1                    NaN    jnardino                 NaN              0   
2                    NaN  yvonnalynn                 NaN              0   
3                    NaN    jnardino                 NaN              0   
4                    NaN    jnardino                 NaN              0   

                                                text tweet_coord  \
0                @VirginAmerica What @dhepburn said.         NaN   
1  @VirginAmerica plus you've added commercials t...         NaN   
2  @VirginAmerica I didn't today... Must mean I n...         NaN   
3  @VirginAmerica it's really aggressive to blast...         NaN   
4  @VirginAmerica and it's a really big bad thing...         NaN   

               tweet_created tweet_location               user_timezone  \
0  2015-02-24 11:35:52 -0800            NaN  Eastern Time (US & Canada)   
1  2015-02-24 11:15:59 -0800            NaN  Pacific Time (US & Canada)   
2  2015-02-24 11:15:48 -0800      Lets Play  Central Time (US & Canada)   
3  2015-02-24 11:15:36 -0800            NaN  Pacific Time (US & Canada)   
4  2015-02-24 11:14:45 -0800            NaN  Pacific Time (US & Canada)   

                                        cleaned_data  
0                   VirginAmerica What dhepburn said  
1  VirginAmerica plus you 've added commercials t...  
2  VirginAmerica I did n't today ... Must mean I ...  
3  VirginAmerica it 's really aggressive to blast...  
4  VirginAmerica and it 's a really big bad thing...

In [ ]:
i = 0

cleaned_data_train_negative = []
cleaned_data_train_positive = []

for data in X_train:
  if y_train.iloc[i] == 'negative':
    data1 = RemoveTokens(data)
    cleaned_data_train_negative.append(data1)
  i+=1

cleaned_data_train = pd.Series(cleaned_data_train_negative)


tokenized_text = [word_tokenize(text) for text in cleaned_data_train_negative]
all_words = [word for sublist in tokenized_text for word in sublist]
s = pd.Series(all_words)
cnt = s.value_counts()
top = cnt.head(15)
print('Negative')
print(top)

Negative
to             4733
I              3494
the            3162
a              2464
and            2191
on             2170
for            2115
united         2088
you            1884
flight         1854
USAirways      1818
#              1808
my             1774
AmericanAir    1679
is             1624
Name: count, dtype: int64


In [ ]:
i = 0

cleaned_data_train_neutral = []

for data in X_train:
  if y_train.iloc[i] == 'neutral':
    data1 = RemoveTokens(data)
    cleaned_data_train_neutral.append(data1)
  i+=1


cleaned_data_train = pd.Series(cleaned_data_train_neutral)


tokenized_text = [word_tokenize(text) for text in cleaned_data_train_neutral]
all_words = [word for sublist in tokenized_text for word in sublist]
s = pd.Series(all_words)
cnt = s.value_counts()
top = cnt.head(15)
print('Neutral')
print(top)

Neutral
to              1336
I               1049
the              745
a                615
you              593
JetBlue          589
united           556
#                541
on               526
SouthwestAir     525
for              480
flight           409
AmericanAir      407
my               407
in               372
Name: count, dtype: int64


In [ ]:
i = 0
cleaned_data_train_neutral = []

for data in X_train:
  if y_train.iloc[i] == 'positive':
    data1 = RemoveTokens(data)
    cleaned_data_train_positive.append(data1)
  i+=1

cleaned_data_train = pd.Series(cleaned_data_train_positive)


tokenized_text = [word_tokenize(text) for text in cleaned_data_train_positive]
all_words = [word for sublist in tokenized_text for word in sublist]
s = pd.Series(all_words)
cnt = s.value_counts()
top = cnt.head(15)
print('Positive')
print(top)

Positive
to              744
the             741
you             645
I               583
#               574
for             542
JetBlue         450
SouthwestAir    448
united          414
a               405
and             357
AmericanAir     272
in              271
on              265
my              261
Name: count, dtype: int64


In [ ]:
from nltk.corpus import stopwords
from matplotlib import pyplot as plt

stop_words = set(stopwords.words('english'))

def RemoveStopWords(text):
  text = word_tokenize(text)
  stop = [word for word in text if word in stop_words]
  text = [word for word in text if word not in stop_words]
  text = " ".join(text)
  stop = " ".join(stop)
  '''
  stop_set = set(stop)
  stop_dict = {}
  for t in stop_set:
    n = stop.count(t)
    stop_dict[t] = n
  print("Stop word frequencies: " + str(stop_dict))

  plt.bar(list(stop_dict.keys()), list(stop_dict.values()))
  plt.xlabel("Stop words")
  plt.ylabel("Frequencies")
  plt.title("Stop Word Frequencies")
  plt.show()
  '''

  return text, stop

In [ ]:
stop_data = []
stop_list = []
for data in cleaned_data:
  data1, stop = RemoveStopWords(data)
  stop_data.append(data1)
  stop_list.append(stop)

stop_data_series = pd.Series(stop_list)

tokenized_text = [word_tokenize(text) for text in stop_data_series]
all_words = [word for sublist in tokenized_text for word in sublist]
s = pd.Series(all_words)
cnt = s.value_counts()
top = cnt.head(15)
print('Stop Words')
print(top)

df['stop_data'] = stop_data


Stop Words
to      8591
the     5797
a       4364
for     3954
you     3932
on      3671
and     3576
my      3041
is      2727
in      2479
of      2104
it      2086
me      1883
have    1651
was     1628
Name: count, dtype: int64


In [ ]:
plane_words = ['americanair', 'united', 'delta', 'southwestair', 'jetblue', 'virginamerica', 'usairways', 'flight', 'plane', 'AmericanAir', 'United', 'Delta', 'SouthwestAir', 'JetBlue', 'VirginAmerica', 'USAirways']
def RemovePlane(text):
  text = word_tokenize(text)
  plane = [word for word in text if word in plane_words]
  text = [word for word in text if word not in plane_words]
  text = " ".join(text)
  plane = " ".join(plane)

  return text, plane

In [ ]:
plane_data = []
plane_list = []
for data in stop_data:
  data1, plane = RemovePlane(data)
  plane_data.append(data1)
  plane_list.append(plane)

plane_data_series = pd.Series(plane_list)

tokenized_text = [word_tokenize(text) for text in plane_data_series]
all_words = [word for sublist in tokenized_text for word in sublist]
s = pd.Series(all_words)
cnt = s.value_counts()
top = cnt.head(15)
print('Plane Words')
print(top)

df['plane_data'] = plane_data

Plane Words
united           3828
flight           3164
USAirways        2935
AmericanAir      2925
SouthwestAir     2407
JetBlue          2286
plane             607
VirginAmerica     501
United            325
usairways         105
jetblue            92
Delta              87
southwestair       46
americanair        37
delta              23
Name: count, dtype: int64


In [ ]:
df.head()

tweet_id airline_sentiment  airline_sentiment_confidence  \
0  570306133677760513           neutral                        1.0000   
1  570301130888122368          positive                        0.3486   
2  570301083672813571           neutral                        0.6837   
3  570301031407624196          negative                        1.0000   
4  570300817074462722          negative                        1.0000   

  negativereason  negativereason_confidence         airline  \
0            NaN                        NaN  Virgin America   
1            NaN                     0.0000  Virgin America   
2            NaN                        NaN  Virgin America   
3     Bad Flight                     0.7033  Virgin America   
4     Can't Tell                     1.0000  Virgin America   

  airline_sentiment_gold        name negativereason_gold  retweet_count  \
0                    NaN     cairdin                 NaN              0   
1                    NaN    jnardino                 NaN              0   
2                    NaN  yvonnalynn                 NaN              0   
3                    NaN    jnardino                 NaN              0   
4                    NaN    jnardino                 NaN              0   

                                                text tweet_coord  \
0                @VirginAmerica What @dhepburn said.         NaN   
1  @VirginAmerica plus you've added commercials t...         NaN   
2  @VirginAmerica I didn't today... Must mean I n...         NaN   
3  @VirginAmerica it's really aggressive to blast...         NaN   
4  @VirginAmerica and it's a really big bad thing...         NaN   

               tweet_created tweet_location               user_timezone  \
0  2015-02-24 11:35:52 -0800            NaN  Eastern Time (US & Canada)   
1  2015-02-24 11:15:59 -0800            NaN  Pacific Time (US & Canada)   
2  2015-02-24 11:15:48 -0800      Lets Play  Central Time (US & Canada)   
3  2015-02-24 11:15:36 -0800            NaN  Pacific Time (US & Canada)   
4  2015-02-24 11:14:45 -0800            NaN  Pacific Time (US & Canada)   

                                        cleaned_data  \
0                   VirginAmerica What dhepburn said   
1  VirginAmerica plus you 've added commercials t...   
2  VirginAmerica I did n't today ... Must mean I ...   
3  VirginAmerica it 's really aggressive to blast...   
4  VirginAmerica and it 's a really big bad thing...   

                                           stop_data  \
0                   VirginAmerica What dhepburn said   
1  VirginAmerica plus 've added commercials exper...   
2  VirginAmerica I n't today ... Must mean I need...   
3  VirginAmerica 's really aggressive blast obnox...   
4              VirginAmerica 's really big bad thing   

                                          plane_data  encoded_sentiments  
0                                 What dhepburn said                   1  
1    plus 've added commercials experience ... tacky                   2  
2  I n't today ... Must mean I need take another ...                   1  
3  's really aggressive blast obnoxious `` entert...                   0  
4                            's really big bad thing                   0

In [ ]:
import sklearn
from sklearn.preprocessing import LabelEncoder

airline_sentiment = df['airline_sentiment']

encoded_sentiments = []

le = LabelEncoder()
encoded = le.fit_transform(airline_sentiment)

df['encoded_sentiments'] = encoded

In [ ]:
df.head()

tweet_id airline_sentiment  airline_sentiment_confidence  \
0  570306133677760513           neutral                        1.0000   
1  570301130888122368          positive                        0.3486   
2  570301083672813571           neutral                        0.6837   
3  570301031407624196          negative                        1.0000   
4  570300817074462722          negative                        1.0000   

  negativereason  negativereason_confidence         airline  \
0            NaN                        NaN  Virgin America   
1            NaN                     0.0000  Virgin America   
2            NaN                        NaN  Virgin America   
3     Bad Flight                     0.7033  Virgin America   
4     Can't Tell                     1.0000  Virgin America   

  airline_sentiment_gold        name negativereason_gold  retweet_count  \
0                    NaN     cairdin                 NaN              0   
1                    NaN    jnardino                 NaN              0   
2                    NaN  yvonnalynn                 NaN              0   
3                    NaN    jnardino                 NaN              0   
4                    NaN    jnardino                 NaN              0   

                                                text tweet_coord  \
0                @VirginAmerica What @dhepburn said.         NaN   
1  @VirginAmerica plus you've added commercials t...         NaN   
2  @VirginAmerica I didn't today... Must mean I n...         NaN   
3  @VirginAmerica it's really aggressive to blast...         NaN   
4  @VirginAmerica and it's a really big bad thing...         NaN   

               tweet_created tweet_location               user_timezone  \
0  2015-02-24 11:35:52 -0800            NaN  Eastern Time (US & Canada)   
1  2015-02-24 11:15:59 -0800            NaN  Pacific Time (US & Canada)   
2  2015-02-24 11:15:48 -0800      Lets Play  Central Time (US & Canada)   
3  2015-02-24 11:15:36 -0800            NaN  Pacific Time (US & Canada)   
4  2015-02-24 11:14:45 -0800            NaN  Pacific Time (US & Canada)   

                                        cleaned_data  \
0                   VirginAmerica What dhepburn said   
1  VirginAmerica plus you 've added commercials t...   
2  VirginAmerica I did n't today ... Must mean I ...   
3  VirginAmerica it 's really aggressive to blast...   
4  VirginAmerica and it 's a really big bad thing...   

                                           stop_data  \
0                   VirginAmerica What dhepburn said   
1  VirginAmerica plus 've added commercials exper...   
2  VirginAmerica I n't today ... Must mean I need...   
3  VirginAmerica 's really aggressive blast obnox...   
4              VirginAmerica 's really big bad thing   

                                          plane_data  encoded_sentiments  
0                                 What dhepburn said                   1  
1    plus 've added commercials experience ... tacky                   2  
2  I n't today ... Must mean I need take another ...                   1  
3  's really aggressive blast obnoxious `` entert...                   0  
4                            's really big bad thing                   0

In [ ]:
from sklearn.feature_extraction.text import CountVectorizer

vectorizer = CountVectorizer()

vectorized_text = vectorizer.fit_transform(df['plane_data'])
#vectorized_text = vectorized_text.toarray()

#vectorized_text = [row for row in vectorized_text]
#df['vectorized'] = vectorized_text

In [ ]:
df.head()

tweet_id airline_sentiment  airline_sentiment_confidence  \
0  570306133677760513           neutral                        1.0000   
1  570301130888122368          positive                        0.3486   
2  570301083672813571           neutral                        0.6837   
3  570301031407624196          negative                        1.0000   
4  570300817074462722          negative                        1.0000   

  negativereason  negativereason_confidence         airline  \
0            NaN                        NaN  Virgin America   
1            NaN                     0.0000  Virgin America   
2            NaN                        NaN  Virgin America   
3     Bad Flight                     0.7033  Virgin America   
4     Can't Tell                     1.0000  Virgin America   

  airline_sentiment_gold        name negativereason_gold  retweet_count  \
0                    NaN     cairdin                 NaN              0   
1                    NaN    jnardino                 NaN              0   
2                    NaN  yvonnalynn                 NaN              0   
3                    NaN    jnardino                 NaN              0   
4                    NaN    jnardino                 NaN              0   

                                                text tweet_coord  \
0                @VirginAmerica What @dhepburn said.         NaN   
1  @VirginAmerica plus you've added commercials t...         NaN   
2  @VirginAmerica I didn't today... Must mean I n...         NaN   
3  @VirginAmerica it's really aggressive to blast...         NaN   
4  @VirginAmerica and it's a really big bad thing...         NaN   

               tweet_created tweet_location               user_timezone  \
0  2015-02-24 11:35:52 -0800            NaN  Eastern Time (US & Canada)   
1  2015-02-24 11:15:59 -0800            NaN  Pacific Time (US & Canada)   
2  2015-02-24 11:15:48 -0800      Lets Play  Central Time (US & Canada)   
3  2015-02-24 11:15:36 -0800            NaN  Pacific Time (US & Canada)   
4  2015-02-24 11:14:45 -0800            NaN  Pacific Time (US & Canada)   

                                        cleaned_data  \
0                   VirginAmerica What dhepburn said   
1  VirginAmerica plus you 've added commercials t...   
2  VirginAmerica I did n't today ... Must mean I ...   
3  VirginAmerica it 's really aggressive to blast...   
4  VirginAmerica and it 's a really big bad thing...   

                                           stop_data  \
0                   VirginAmerica What dhepburn said   
1  VirginAmerica plus 've added commercials exper...   
2  VirginAmerica I n't today ... Must mean I need...   
3  VirginAmerica 's really aggressive blast obnox...   
4              VirginAmerica 's really big bad thing   

                                          plane_data  encoded_sentiments  \
0                                 What dhepburn said                   1   
1    plus 've added commercials experience ... tacky                   2   
2  I n't today ... Must mean I need take another ...                   1   
3  's really aggressive blast obnoxious `` entert...                   0   
4                            's really big bad thing                   0   

                                          vectorized  
0  <Compressed Sparse Row sparse matrix of dtype ...  
1  <Compressed Sparse Row sparse matrix of dtype ...  
2  <Compressed Sparse Row sparse matrix of dtype ...  
3  <Compressed Sparse Row sparse matrix of dtype ...  
4  <Compressed Sparse Row sparse matrix of dtype ...

In [ ]:
y = df['encoded_sentiments']
X = vectorized_text
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


In [ ]:
from sklearn.metrics import confusion_matrix
from sklearn.svm import SVC

svm_model_linear = SVC(kernel = 'linear', C = 1).fit(X_train, y_train)
svm_predictions = svm_model_linear.predict(X_test)



In [ ]:
accuracy = svm_model_linear.score(X_test, y_test)


In [ ]:
cm = confusion_matrix(y_test, svm_predictions)

In [ ]:
accuracy

0.7681010928961749

In [ ]:
cm

array([[1620,  203,   66],
       [ 211,  309,   60],
       [  90,   49,  320]])

True Negative(actual: negative, predicted: negative) = 1620

False Neutral(actual: negative, predicted: neutral) = 203

False Positive(actual: negative, predicted: positive) = 66

False Negative(actual: neutral, predicted: negative) = 211

True Neutral(actual: neutral, predicted: neutral) = 309

False Positive(actual: neutral, predicted: positive) = 60

False Negative(actual: positive, predicted: negative) = 90

False Neutral(actual: positive, predicted: neutral) = 49

True Positive(actual: positive, predicted: positive) = 320




